jupyterdrive/gdrive/drive-contents.js

```diff
@@ -87,6 +52,136 @@ define(function(require) {
     };
 
     /**
+     * A utility class to detect cloberring of files by other users
+     */
+    var ClobberDetector = function() {
+        /**
+         * Stores the last few UUIDs that were attached to the last save
+         * for a given file.  This is used when checking if a file has been
+         *  modified by another user.
+         */
+        this.save_uuids = {};
+    };
+
+    /**
+     * @type {String}
+     * A UUID used to detect when other users overwrite saves done by
+     * this client.  If this property of the file changes, then
+     * the user is warned that another user is also saving to the file
+     * and that user's work will be overwritten.
+     */
+    ClobberDetector.SAVE_UUID_KEY = 'save_uuid';
+
+    /**
+     * @type {Number}
+     * The number of UUIDs per file to keep track of.
+     */
+    ClobberDetector.NUM_UUIDS_IN_CACHE = 100;
+
+    ClobberDetector.prototype.add_uuid_to_cache = function(resource, uuid) {
+        var file_id = resource['id'];
+        this.save_uuids[file_id] = this.save_uuids[file_id] || [];
+        var uuids = this.save_uuids[file_id];
+        uuids.push(uuid);
+        if (uuids.length > ClobberDetector.NUM_UUIDS_IN_CACHE) {
+            uuids.shift();
+        }
+        console.log(uuids);
+    }
+
+    /**
+     * Returns a UUID
+     * @return {string}
+     */
+    ClobberDetector.prototype.create_uuid = function() {
+        return Math.random().toString().substr(2);
+    }
```

Comment:
if you want uuid that reflect http://www.ietf.org/rfc/rfc4122.txt , `services/utils` shoudl have such a generator.